src/Avalonia.Controls/Expander.cs

```diff
@@ -332,5 +288,63 @@ private void UpdatePseudoClasses()
 
             PseudoClasses.Set(":expanded", IsExpanded);
         }
+
+        /// <summary>
+        /// Called when the <see cref="IsExpanded"/> property has to be coerced.
+        /// </summary>
+        /// <param name="value">The value to coerce.</param>
+        protected virtual bool OnCoerceIsExpanded(bool value)
+        {
+            RoutedEventArgs eventArgs;
```

Comment:
Also note I kept using `RoutedEventArgs` here. The other branch created a new `CancelRoutedEventArgs` class with a `Cancel` property (similar to [CancelEventArgs](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.canceleventargs?view=net-7.0)). Personally, I don't think that is needed; however, if others do please speak up.

Reply:
I added the explicit cancel property on my branch partly so that this and `SplitView` used the same event args, but also because marking the event as handled to cancel the property change is is blurring two separate concepts together.

If a handler sets `e.Handled = true`, then no further event handlers see it (unless the subscribed with `handledEventsToo`, but that is a special case).

Now, what if a control wants to block a parent control from cancelling the change? This isn't possible. Either the control cancels the change itself, or it allows the event to pass on to the parent, which will cancel it.

That scenario won't be common, but it is something that could reasonably happen in someone's `UserControl`.

Reply:
As far as I am aware, no `CancelRoutedEventArgs` has ever existed from WPF or in later XAML frameworks. `Handled = true` is used to effectively cancel the event. Yes, this may stop propagation in most cases as well but it hasn't historically been an issue in the core frameworks... that said an internet search will show several control libraries have implemented `CancelRoutedEventArgs` in various places. 

You also make some good points that there are slight differences between marking the event as handled and marking it as cancelled. Maybe uncancelling an event is also something we should support...

With those two points in mind, I'll add the class unless one of the core maintainers disagrees.

Reply:
Seems everyone is OK with this and it is preferred so marking resolved.

Linking to other conversions:

 * https://github.com/AvaloniaUI/Avalonia/pull/9555#discussion_r1033128176
 * https://github.com/AvaloniaUI/Avalonia/pull/9555#issuecomment-1383758760